csharp/src/Microsoft.ML.OnnxRuntime/ProviderOptions.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// Holds the options for configuring a CUDA Execution Provider instance
+    /// </summary>
+    public class OrtCUDAProviderOptions : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructs an empty OrtCUDAProviderOptions instance
+        /// </summary>
+        public OrtCUDAProviderOptions() : base(IntPtr.Zero, true)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateCUDAProviderOptions(out handle));
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Updates  the configuration knobs of OrtCUDAProviderOptions that will eventually be used to configure a CUDA EP
+        /// Please refer to the following on different key/value pairs to configure a CUDA EP and their meaning:
+        /// https://github.com/microsoft/onnxruntime/blob/gh-pages/docs/reference/execution-providers/CUDA-ExecutionProvider.md
+        /// </summary>
+        /// <param name="keys">keys of all the configuration knobs of a CUDA Execution Provider</param>
+        /// <param name="values">values of all the configuration knobs of a CUDA Execution Provider (must match number of keys)</param>
+
+        public void UpdateOptions(string[] keys, string[] values)
```

Comment:
>string[] keys, string[] values [](start = 34, length = 30)

would a Dictionary<string, string> be a friendlier way to pass the config options?